test/tests/unit/replset/read_preference_tests.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const expect = require('chai').expect;
+const ReplSet = require('../../../../lib/topologies/replset');
+const ReadPreference = require('../../../../lib/topologies/read_preference');
+const mock = require('../../../mock');
+const ReplSetFixture = require('../common').ReplSetFixture;
+
+describe('Secondaries (ReplSet)', function() {
+  let test;
+  before(() => (test = new ReplSetFixture()));
+  afterEach(() => mock.cleanup());
+  beforeEach(() => test.setup());
+
+  it('Should not be "connected" with ReadPreference secondary unless secondary is connected', {
+    metadata: {
+      requires: {
+        topology: 'single'
+      }
+    },
+
+    test: function(done) {
+      const replSet = new ReplSet(
+        [test.primaryServer.address(), test.firstSecondaryServer.address()],
+        {
+          setName: 'rs',
+          connectionTimeout: 3000,
+          socketTimeout: 0,
+          haInterval: 100,
+          size: 1
+        }
+      );
+
+      replSet.on('error', done);
+
+      replSet.on('connect', server => {
+        let err;
+        try {
```

Comment:
do we actually need this try/catch? if the assertion fails we will still get the message during the test run (e.g. I think there is already a try/catch internal to mocha that provides this same behavior)

Reply:
since it is asynchronous, it won't work